neo4j-jdbc/src/main/java/org/neo4j/jdbc/StatementImpl.java

```diff
@@ -157,6 +157,7 @@ public void close() throws SQLException {
 		}
 		closeResultSet();
 		this.closed = true;
+		this.connection.onClose(this);
```

Comment:
As the connection hands out the statement, it should keep track of it. If this is not possible, define a callback independent of the connection and don't intertwine both concepts. 

I think adding an `addCloseListenener` on the statementimpl class is more appropriate.

Reply:
Updated as agreed.